### PR TITLE
Add Support for AWS Cost Explorer's withGroupBy Feature

### PR DESCRIPTION
**Pull request by Pasukaru1996**
This pull request adds withGroupBy support from AWS Cost Explorer and adds pagination support.

**Key Highlights:**

- Added `withGroupBy` support for AWS Cost Explorer.
- Implemented pagination support.

**Additional Improvements:**

- Handled empty array responses for groups with `withGroupBy`.
- Updated plugin version to 0.0.5.
- Updated plugin documentation.
- Dynamic addition of `group_key1` and `group_key2` columns when `withGroupBy` is specified.
- Added configuration options and validations for groups.
- Updated `aws-java-sdk-costexplorer` to version 1.12.666.
- Performed code cleanup.

### CODE REVIEW

File: src/main/java/org/embulk/input/aws_cost_explorer/client/AwsCostExplorerClient.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                if (isFinished) {
+                    return false;
+                }
+
+                final AwsCostExplorerResponse response = request(requestParameters);
+                action.accept(response);
+
+                prepareNextRequest(response);
+
+                return !isFinished;
```
Comment (chikamura) [inline src/main/java/org/embulk/input/aws_cost_explorer/client/AwsCostExplorerClient.java]: nits: The isFinished variable may be deleted. (If it's an idiom, ignore it.)

```suggestion
                final AwsCostExplorerResponse response = request(requestParameters);
                action.accept(response);

                final String nextPageToken = response.getNextPageToken();
                if (nextPageToken == null || nextPageToken.isEmpty()) {
                    return false;
                }
                requestParameters.setNextPageToken(nextPageToken);

                return true;
```

File: src/main/java/org/embulk/input/aws_cost_explorer/client/AwsCostExplorerResponse.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            if (resultsByTime.getGroups().isEmpty()) {
+                addRecordToPageWithoutGroupBy(pageBuilder, task, resultsByTime);
+            }
+            else {
+                addRecordsToPageWithGroupBy(pageBuilder, task, resultsByTime);
+            }
+        });
+    }
+
+    private void addRecordToPageWithoutGroupBy(PageBuilder pageBuilder, PluginTask task, ResultByTime resultsByTime)
```
Comment (chikamura) [inline src/main/java/org/embulk/input/aws_cost_explorer/client/AwsCostExplorerResponse.java:62]: nits: There is a duplication in addRecordsToPageWithGroupBy, which may need to be refactored if there is time.

**Review by chikamura — APPROVED** [review]
LGTM

**Review by d-hrs — APPROVED** [review]
LGTM

**Pull request merged**